### ISSUE TITLE
Use std streams instead of Tempfile?

### ISSUE THREAD

**Issue by YurySolovyov** [documentcloud/closure-compiler#43]
Basically it should be possible to just write something into stdin via echo:
```
> f = `echo "const foo = () => { return 4 };" | java -jar lib/closure-compiler-20170521.jar`
...
> f
 => "var foo=function(){return 4};\n"
```

**Comment by knowtheory** [comment]
Yeah, this seems like a reasonable way to go.  Being able to pipe the output somewhere for inspection might be handy, but the test suite should also raise all of that in a reasonable fashion anyway.

**Comment by YurySolovyov** [comment]
I was thinking about this more in terms of perf. gains, not a user-facing feature.

**Comment by YurySolovyov** [comment]
I've been playing with it a bit, it is possible to do that even without `echo`:
```ruby
require 'open3'

command = "java -jar lib/closure-compiler-20170626.jar"
stdin, stdout, stderr, wait_thr = Open3::popen3(command)
stdin.write("const foo = () => { return { foo: 321, bar: ['a', 'b', 'c'] } };")
stdin.close
stdout.read
```
gives me 
```
=> "var foo=function(){return{foo:321,bar:[\"a\",\"b\",\"c\"]}};\n" 
```

**Comment by YurySolovyov** [comment]
also, it seems like `-XX:TieredStopAtLevel=1` CLI flag reduces startup time significantly
